Revert to Rust Givens rotations implementation after resolving NaN issue

On some machines it results in NaN, while on others it's not NaN but the vector is incorrect and unnormalized. Switching from the Rust code to the Python code seems to fix it.

```python
import numpy as np
import pyscf

import ffsim

mol = pyscf.gto.Mole()
mol.build(
    atom=[["N", (0, 0, 0)], ["N", (1.0, 0, 0)]],
    basis="sto-6g",
    symmetry="d2h",
)
active_space = range(2, mol.nao_nr())
mol_data = ffsim.MolecularData.from_mole(mol, active_space=active_space, ccsd=True)
norb = mol_data.norb
nelec = mol_data.nelec
t1 = mol_data.ccsd_t1
t2 = mol_data.ccsd_t2

# get orbital rotation
op = ffsim.UCJOperator.from_t_amplitudes(t2, t1_amplitudes=t1)
orbital_rotation = op.orbital_rotations[0]
assert ffsim.linalg.is_unitary(orbital_rotation)

# create state vector
vec = ffsim.hartree_fock_state(norb, nelec)
np.testing.assert_allclose(np.linalg.norm(vec), 1)

# apply orbital rotation to state vector
statevec = ffsim.apply_orbital_rotation(vec, orbital_rotation, norb, nelec)
np.testing.assert_allclose(np.linalg.norm(statevec), 1)
```

```
Traceback (most recent call last):
  File "/home/kjs/projects/ffsim/scratch/orbital_rotation_bug.py", line 30, in <module>
    np.testing.assert_allclose(np.linalg.norm(statevec), 1)
  File "/home/kjs/.local/share/virtualenvs/ffsim-1cfkSnAR/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 1508, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/home/kjs/.pyenv/versions/3.10.12/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/kjs/.local/share/virtualenvs/ffsim-1cfkSnAR/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 722, in assert_array_compare
    flagged = func_assert_same_pos(x, y, func=isnan, hasval='nan')
  File "/home/kjs/.local/share/virtualenvs/ffsim-1cfkSnAR/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 692, in func_assert_same_pos
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

x and y nan location mismatch:
 x: array(nan)
 y: array(1)
```

--- a/python/ffsim/gates/orbital_rotation.py
+++ b/python/ffsim/gates/orbital_rotation.py
@@ -21,14 +21,9 @@ from pyscf.fci import cistring
 from scipy.special import comb
 
 from ffsim._lib import (
-    # apply_givens_rotation_in_place,
+    apply_givens_rotation_in_place,
     apply_phase_shift_in_place,
     apply_single_column_transformation_in_place,
-)
-
-# TODO delete this and uncomment import from ffsim._lib after fixing bug
-from ffsim._slow.gates.orbital_rotation import (
-    apply_givens_rotation_in_place_slow as apply_givens_rotation_in_place,
 )
 from ffsim.cistring import gen_orbital_rotation_index
 from ffsim.linalg import givens_decomposition, lup

--- a/src/gates/orbital_rotation.rs
+++ b/src/gates/orbital_rotation.rs
@@ -160,7 +160,8 @@ pub fn apply_givens_rotation_in_place(
     let shape = vec.shape();
     let dim_b = shape[1] as i32;
     let s_abs = s.norm();
-    let phase = s / s_abs;
+    let angle = s.arg();
+    let phase = Complex64::new(angle.cos(), angle.sin());
     let phase_conj = phase.conj();
 
     // TODO parallelize this

